### DISCUSSION TITLE
Send iBeacons id to Home Assistant

### DISCUSSION THREAD

**Discussion by LeeWiu** [arendst/Tasmota#22355] [Q&A]
I flashed tasmota32-bluetooth.bin v14.3.0 on Shelly Plus2PM and trying to setup ibeacon tracker
```
so115 1
ibeacon 1
BLEAddrFilter 3
BLEAlias 5D8A03803F44 MyBeacon1
BLEAlias 67093FE8A404 MyBeacon2
iBeaconOnlyAliased 1
```
I can see in console:

> 17:09:20.001 iBeacon: MAC: 67:09:3f:e8:a4:04 Major: 25600 Minor: 17564 UUID: 08ac3a89-02f4-43a2-a7b0-f212ea0dc4b4 Power: 197 RSSI: -80
17:09:20.146 MQT: tele/livingroom_light/SENSOR = {"Time":"2024-10-24T17:09:20","IBEACON":{"UID":"08AC3A8902F443A2A7B0F212EA0DC4B4","MAJOR":"0064","MINOR":"9C44","MAC":"67093FE8A404","RSSI":-80,"STATE":"ON","PERSEC":0}}
17:11:31.034 iBeacon: MAC: 6e:98:c0:2f:b1:d4 Major: 25600 Minor: 17564 UUID: 94221ff6-adf7-46ae-b48f-c845d908b23b Power: 197 RSSI: -95
17:11:31.086 MQT: tele/livingroom_light/SENSOR = {"Time":"2024-10-24T17:11:31","IBEACON":{"UID":"94221FF6ADF746AEB48FC845D908B23B","MAJOR":"0064","MINOR":"9C44","MAC":"6E98C02FB1D4","RSSI":-95,"STATE":"ON","PERSEC":0}}

I added iBeacon integration in Home Assistant and added the 2 beacons to allowed UUIDs.
What next? What commands I need to send in Tasmota console?
If I add a esphome bluetooth-proxy, the ibeacons works great but I want to use only Shelly devices.

**Comment by sfromis** [comment]
Tasmota is already publishing messages including data about the beacons. Maybe you could have a setup in HA able to subscribe to those messages, or you could have Tasmota send data in another format, probably using the Berry language providing good features for local programming. Details would depend on how that HA iBeacon integration works, and how it can pick up data.

**Comment by LeeWiu** [comment]
Anyone can help me with a berry script please?

**Comment by LeeWiu** [comment]
I forget to ask, I only tried bluetooth precompiled version, this option is already enabled? How to check?
![Screenshot_20241101-195559](https://github.com/user-attachments/assets/30d84812-64b0-47e8-8ae1-8aa829d525c2)

**Reply by sfromis** [reply]
Well, the screenshot is very explicit about beacon support not included in precompiled binaries. Hence not in the one you used.

Still, you can check by executing the `iBeacon` command, if it gives an error or a result.

**Comment by LeeWiu** [comment]
They are already detected but not seen by HA:

> 22:56:27.655 CMD: Grp 0, Cmd 'IBEACON', Idx 1, Len 0, Pld -99, Data ''
> 22:56:27.666 MQT: stat/livingroom_light/RESULT = {"IBEACON":1}
> 22:56:48.878 MQT: tele/livingroom_light/SENSOR = {"Time":"2024-11-01T22:56:48","IBEACON":{"UID":"94221FF6ADF746AEB48FC845D908B23B","NAME":"MyBeacon3","MAJOR":"0064","MINOR":"9C44","MAC":"64C2FE4F19C8","RSSI":0,"STATE":"OFF","PERSEC":0}}
> 22:56:53.897 iBeacon: MAC: 64:c2:fe:4f:19:c8 Major: 25600 Minor: 17564 UUID: 94221ff6-adf7-46ae-b48f-c845d908b23b Power: 197 RSSI: -93
> 22:56:53.914 MQT: tele/livingroom_light/SENSOR = {"Time":"2024-11-01T22:56:53","IBEACON":{"UID":"94221FF6ADF746AEB48FC845D908B23B","NAME":"MyBeacon3","MAJOR":"0064","MINOR":"9C44","MAC":"64C2FE4F19C8","RSSI":-93,"STATE":"ON","PERSEC":0}}

**Reply by sfromis** [reply]
Well, seeing them in the console does not mean that HA will have code to use those MQTT payloads. You'll likely have to do some manual configuration, unless you can install something "knowing" about beacons already.

**Comment by LeeWiu** [comment]
So "define USE_IBEACON_ESP32 " it's already enabled in precompiled tasmota32-bluetooth.bin if the ibeacons are detected?
This integration it's not enough to detect them? https://www.home-assistant.io/integrations/ibeacon

**Reply by sfromis** [reply]
Not using HA, I cannot answer about details, but as that page does not mention "Tasmota", my guess would be "no".
